pkg/ebpftracer/c/tracee.bpf.c

```diff
@@ -2719,8 +2761,123 @@ int BPF_KPROBE(trace_security_socket_connect)
         stsb(args_buf, (void *) address, sockaddr_len, 2);
     }
 
-    // Submit the event.
-    return events_perf_submit(&p, SECURITY_SOCKET_CONNECT, 0);
+    return events_perf_submit(p, SECURITY_SOCKET_CONNECT, 0);
+}
+
+statfunc int send_stdio_via_socket_from_sock_connect(struct pt_regs *ctx, program_data_t *p) {
+    u64 addr_len = PT_REGS_PARM3(ctx);
+
+    struct socket *sock = (struct socket *) PT_REGS_PARM1(ctx);
+    if (!sock)
+        return 0;
+
+    struct sockaddr *address = (struct sockaddr *) PT_REGS_PARM2(ctx);
+    if (!address)
+        return 0;
+
+    // Check if the socket type is supported.
+    u32 type = BPF_CORE_READ(sock, type);
+    switch (type) {
+        // TODO: case SOCK_DCCP:
+        case SOCK_DGRAM:
+        case SOCK_SEQPACKET:
+        case SOCK_STREAM:
+            break;
+        default:
+            return 0;
+    }
+
+    // Check if the socket family is supported.
+    sa_family_t sa_fam = get_sockaddr_family(address);
+    switch (sa_fam) {
+        case AF_INET:
+        case AF_INET6:
```

Comment:
This cannot be right, as in C switch cases fall through, meaning we would always `return 0` here.